### PR TITLE
Add Projector Obs to LGPU

### LINKED ISSUE #883: “Observable Projector not supported on device lightning.gpu”
# I find “Observable Projector not supported on device lightning.gpu”, but it's said to be supported in the stable version document.

### When I run the following codes, there will be no error.
```
dev = qml.device("lightning.gpu", wires=num_wires)
wires = list(range(num_wires))
@qml.qnode(dev, diff_method="adjoint")
def kernel(x1, x2, params):
    ansatz(x1, params, wires)
    qml.adjoint(ansatz)(x2, params, wires)
    return qml.expval(qml.Projector([0]*num_wires, wires=wires))
```
### But when I try to get the result, it will output “Observable Projector not supported on device lightning.gpu”
```
result = kernel(X1, X2, params)
```
### Here is the device information:

Name: PennyLane
Version: 0.37.0
Summary: PennyLane is a cross-platform Python library for quantum computing, quantum machine learning, and quantum chemistry. Train a quantum computer the same way as a neural network.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /root/anaconda3/envs/pennylane_gpu/lib/python3.11/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, packaging, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml, typing-extensions
Required-by: PennyLane_Lightning, PennyLane_Lightning_GPU

Platform info:           Linux-5.4.0-144-generic-x86_64-with-glibc2.31
Python version:          3.11.9
Numpy version:           1.26.4
Scipy version:           1.13.1
Installed devices:
- lightning.qubit (PennyLane_Lightning-0.37.0)
- default.clifford (PennyLane-0.37.0)
- default.gaussian (PennyLane-0.37.0)
- default.mixed (PennyLane-0.37.0)
- default.qubit (PennyLane-0.37.0)
- default.qubit.autograd (PennyLane-0.37.0)
- default.qubit.jax (PennyLane-0.37.0)
- default.qubit.legacy (PennyLane-0.37.0)
- default.qubit.tf (PennyLane-0.37.0)
- default.qubit.torch (PennyLane-0.37.0)
- default.qutrit (PennyLane-0.37.0)
- default.qutrit.mixed (PennyLane-0.37.0)
- default.tensor (PennyLane-0.37.0)
- null.qubit (PennyLane-0.37.0)
- lightning.gpu (PennyLane_Lightning_GPU-0.37.0)

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Add support for `Projector` observable via diagonalization to LGPU. 

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

Fixes #883 
[sc-72652]

### CODE REVIEW

**Review by vincentmr — COMMENTED** [review]
Nice @maliasadi . Could we just remove this [condition](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/tests/test_expval.py#L109) and instead filter only on the device name (like LTensor)? I think it would make it more obvious whether the device should support the feature or not.

**Comment by maliasadi** [comment]
@vincentmr Nice catch! We can actually remove this condition from those unit tests.

File: tests/test_expval.py [hunk 1]
```diff
@@ -106,9 +106,6 @@ def test_projector_expectation(self, theta, phi, qubit_device, tol):
         dev_def = qml.device("default.qubit", wires=n_qubits)
         dev = qubit_device(wires=n_qubits)
 
-        if "Projector" not in dev.observables:
```
Comment (multiphaseCFD) [inline tests/test_expval.py:109]: This could impact `ltensor`.

**Review by multiphaseCFD — COMMENTED** [review]
Thanks @maliasadi ! Looks good to me.

**Review by multiphaseCFD — APPROVED** [review]
Nice work! Thanks @maliasadi !

**Review by vincentmr — APPROVED** [review]
LGTM, nice short and sweet change. Thanks @maliasadi .

**Pull request merged**